include/openthread/coap.h

```diff
@@ -747,7 +747,7 @@ const otCoapOption *otCoapOptionIteratorGetNextOption(otCoapOptionIterator *aIte
  *
  * @see otCoapMessageAppendUintOption
  */
-otError otCoapOptionIteratorGetOptionUintValue(otCoapOptionIterator *aIterator, uint64_t *const aValue);
+otError otCoapOptionIteratorGetOptionUintValue(otCoapOptionIterator *aIterator, uint64_t *aValue);
```

Comment:
should we also change the implementation signature ([here](https://github.com/openthread/openthread/blob/12b967cce2e06840dd9bccc67971b617c8f43a3c/src/core/api/coap_api.cpp#L207)) to keep them consistent